pkg/profiles/profiles.go

```diff
@@ -298,3 +308,27 @@ func ValidateResponseMatch(rspMatch *sp.ResponseMatch) error {
 
 	return nil
 }
+
+func BuildConfig(namespace, service, controlPlaneNamespace string) *ProfileTemplateConfig {
```

Comment:
sorry i didn't catch this earlier... since `BuildConfig` is only ever called immediately before `RenderProfileTemplate`, consider removing it, and also making `ProfileTemplateConfig` private...

```go
func RenderProfileTemplate(namespace, service, controlPlaneNamespace string, w io.Writer) error {
	config := &profileTemplateConfig{
		ControlPlaneNamespace: controlPlaneNamespace,
		ServiceNamespace:      namespace,
		ServiceName:           service,
		ClusterZone:           "svc.cluster.local",
	}

	template, err := template.New("profile").Parse(Template)
	if err != nil {
		return err
	}
	buf := &bytes.Buffer{}
	err = template.Execute(buf, config)
	if err != nil {
		return err
	}

	_, err = w.Write(buf.Bytes())
	return err
}
```

Reply:
do you think it's worth keeping them separate so we can integrate validation into the `BuildConfig` function, and only pass in validated template params to `RenderProfileTemplate`? That would be the only reason I would have for not integrating it

Reply:
i think if we found that building/validating the template struct became complex enough that it warranted its own function (and testing), it would make sense to separate it out. until we do those things though, i'd tend to keep the code simpler. all that said, i don't feel too strongly here, totally ok to ship either way.

Reply:
done!